util/src/main/java/io/kubernetes/client/util/WebSocketStreamHandler.java

```diff
@@ -0,0 +1,143 @@
+
+    @Override
+    public void open(String protocol, Closeable closer) {
+        this.protocol = protocol;
+        this.closer = closer;
+    }
+        
+    @Override
+    public void bytesMessage(InputStream in) {
+        try {
+            OutputStream out = getOutputStream(in.read());
+            byte[] buffer = new byte[4096];
+            for (int read = in.read(buffer); read != -1; read = in.read(buffer)) {
+                out.write(buffer, 0, read);
+            }
+        } catch (IOException ex) {
+            // TODO use a logger here.
+            ex.printStackTrace();
+        }
+    }
+        
+    @Override
+    public void textMessage(Reader in) {
+        try {
+            // TODO: there has to be a better way to do this...
+            OutputStream out = getOutputStream(in.read());
+            char[] buffer = new char[4096];
+            for (int read = in.read(buffer); read != -1; read = in.read(buffer)) {
+                String data = new String(buffer, 0, read);
+                out.write(data.getBytes("UTF-8"));
+            }
+        } catch (IOException ex) {
+            // TODO use a logger here
+            ex.printStackTrace();
+        }
+    }
+        
+    @Override
+    public void close() {
+        for (PipedOutputStream out: output.values()) {
+            try {
+                out.close();
+            } catch (IOException ex) {
+                // TODO use a logger here
+                ex.printStackTrace();
+            }
+        }
+        for (PipedInputStream in: input.values()) {
+            try {
+                in.close();
+            } catch (IOException ex) {
+                // TODO use a logger here
+                ex.printStackTrace();
+            }
+        }
+        try {
+            closer.close();
+        } catch (IOException ex) {
+            // TODO use a logger here
+            ex.printStackTrace();
+        }
+    }
+
+    /**
+     * Get a specific input stream using its identifier
+     * @param stream The stream to return
+     * @returns The specified stream.
+     */
+    public synchronized InputStream getInputStream(int stream) {
+        if (!input.containsKey(stream)) {
+            try {
+                PipedInputStream pipeIn = new PipedInputStream();
+                PipedOutputStream pipeOut = new PipedOutputStream(pipeIn);
+                output.put(stream, pipeOut);
+                input.put(stream, pipeIn);
+            } catch (IOException ex) {
+                // This is _very_ unlikely, as it requires the above constructor to fail.
+                // don't force callers to catch, but still throw
+                throw new IllegalStateException(ex);
+            }
+        }
+        return input.get(stream);
+    }
+
+    /**
+     * Get a specific output stream using its identifier
+     * @param stream The stream to return
+     * @returns The specified stream.
+     */
+    public synchronized OutputStream getOutputStream(int stream) {

```

Comment:
This is fixed now.  `stdin` was never really implemented before.   Now it is, please take another look.